backend/business-partner-agent/src/main/java/org/hyperledger/bpa/model/BPAProofTemplate.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2020-2021 - for information on the respective copyright owner
+ * see the NOTICE file and/or the repository at
+ * https://github.com/hyperledger-labs/business-partner-agent
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hyperledger.bpa.model;
+
+import io.micronaut.core.annotation.Introspected;
+import io.micronaut.data.annotation.AutoPopulated;
+import io.micronaut.data.annotation.TypeDef;
+import io.micronaut.data.model.DataType;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.hyperledger.bpa.controller.api.prooftemplates.ProofTemplate;
+import org.hyperledger.bpa.impl.verification.prooftemplates.ValidAttributeGroup;
+
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.Table;
+import javax.validation.Valid;
+import javax.validation.constraints.NotEmpty;
+import java.util.UUID;
+import java.util.stream.Stream;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+@Introspected
+@Entity
+@Table(name = "bpa_proof_template")
+public class BPAProofTemplate {
+    @Id
+    @AutoPopulated
+    UUID id;
+
+    @NotEmpty
+    String name;
+
+    @NotEmpty
+    @Valid
+    @ValidAttributeGroup
+    @TypeDef(type = DataType.JSON)
+    BPAAttributeGroups attributeGroups;
```

Comment:
Why not directly use `List<BPAAttributeGroup>`?

Reply:
Probably because of the serialization issue? If so can you comment this either here or in the attribute goup?

Reply:
I will add that comment. Good point.